File: elastic-galaxy/src/main/scala/com/galaxy/elastic/conf/SwaggerConfig.scala
package com.galaxy.elastic.conf

import org.springframework.context.annotation.{Bean, Configuration}
import springfox.documentation.builders.{ApiInfoBuilder, PathSelectors, RequestHandlerSelectors}
import springfox.documentation.spi.DocumentationType
import springfox.documentation.spring.web.plugins.Docket
import springfox.documentation.swagger2.annotations.EnableSwagger2

/**
  * Created by wangpeng
  * Date: 2018/8/3
  * Time: 11:47
  */
@Configuration
@EnableSwagger2
class SwaggerConfig {
  @Bean
  def createRestApi: Docket = new Docket(DocumentationType.SWAGGER_2).apiInfo(apiInfo).select.apis(RequestHandlerSelectors.basePackage("com.galaxy.elastic.controller")).paths(PathSelectors.any).build

  private def apiInfo = new ApiInfoBuilder().title("Elastic").description("Elastic-Galaxy API").version("1.0").build
}

File: dubbo-consumer/src/main/java/com/galaxy/dubbo/consumer/conf/SwaggerConfig.java
package com.galaxy.dubbo.consumer.conf;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.builders.ApiInfoBuilder;
import springfox.documentation.builders.PathSelectors;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.service.ApiInfo;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

/**
 * Swagger配置类
 * <p>
 * Created by wangpeng
 * Date: 2018/9/19
 * Time: 22:59
 */
@Configuration
@EnableSwagger2
public class SwaggerConfig {

    @Bean
    public Docket createRestApi() {
        ApiInfo apiInfo = new ApiInfoBuilder().title("Dubbo").description("Dubbo-Consumer API").version("1.0").build();
        return new Docket(DocumentationType.SWAGGER_2).apiInfo(apiInfo).select().apis(RequestHandlerSelectors.basePackage("com.galaxy.dubbo.consumer.controller")).paths(PathSelectors.any()).build();
    }
}
